src/DataProtection/DataProtection/src/Resources.resx

```diff
@@ -190,6 +190,9 @@
   <data name="KeyRingProvider_DefaultKeyRevoked" xml:space="preserve">
     <value>The key ring's default data protection key {0:B} has been revoked.</value>
   </data>
+  <data name="KeyRingProvider_RefreshFailedOnOtherThread" xml:space="preserve">
+    <value>Another thread failed to refresh the key ring.</value>
```

Comment:
Code comments say that the other thread will provide a better exception. Should this exception say that as well so it is clear to the user?

Reply:
Say more?  My thinking was that you'd get nine "another thread failed" messaged and one actual error, so things would be reasonably clear.  What would you add?  "Find the log message from that thread"?

Reply:
Yep, something that might be helpful here is an indication of what type of log message I should be looking for that is related to this exception.

Reply:
If I knew what to look for, I'd put it here. 😝